DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs

```diff
@@ -245,6 +261,34 @@ private void MultiplayerName_RightClick(object sender, MultiplayerNameRightClick
 
         private void TunnelHandler_CurrentTunnelPinged(object sender, EventArgs e) => UpdatePing();
 
+        private void GameHostActivityTimer_TimeElapsed(object sender, EventArgs e) => gameHostInactiveCheck.CheckHostIsActive();
+
+        private void GameHostInactiveCheck_SendInactiveGameWarningMessage(object sender, EventArgs e)
+        {
+            XNAMessageBox hostInactiveWarningMessageBox = new XNAMessageBox(
+                WindowManager,
+                ClientConfiguration.Instance.InactiveHostWarningTitle,
+                ClientConfiguration.Instance.InactiveHostWarningMessage,
+                XNAMessageBoxButtons.OK
+            );
+            hostInactiveWarningMessageBox.OKClickedAction = InactiveMessageBox_OKClicked;
+            hostInactiveWarningMessageBox.Show();
+        }
+
+        private void GameHostInactiveCheck_CloseInactiveGame(object sender, EventArgs e) => LeaveGameLobby();
+
+        private void InactiveMessageBox_OKClicked(XNAMessageBox messageBox) => gameHostInactiveCheck.HostIsAlive();
+
+        public override void OnMouseMove()
```

Comment:
Wouldn't it be better to subscribe to an event if such exists?

Reply:
> Wouldn't it be better to subscribe to an event if such exists?

don't follow this one sorry, could you give an example?

added the L10N change

Reply:
I am not sure anymore, been too long lol